src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -158,12 +158,15 @@ private[xml] object StaxXmlParser extends Serializable {
     while (!shouldStop) {
       parser.nextEvent match {
         case e: StartElement =>
-          keys += e.getName.getLocalPart
-          values += convertField(parser, valueType, options)
+          try {
+            keys += e.getName.getLocalPart
+            values += convertField(parser, valueType, options)
+          } catch {
+            case NonFatal(_) if options.parseMode == PermissiveMode => // do nothing
```

Comment:
These try-catch blocks are the real change, in addition to undoing the change that handles all casts errors as null